lib/src/core/puzzle_animator.dart

```diff
@@ -14,29 +14,29 @@ class PuzzleAnimator implements PuzzleProxy {
   final List<Body> _locations;
   final _controller = StreamController<PuzzleEvent>();
   bool _nextRandomVertical = true;
-  Puzzle _puzzle;
+  Puzzle? _puzzle;
   int _clickCount = 0;
 
-  bool _stable;
+  bool? _stable;
```

Comment:
You could make this non-null and initialize to `true`:

```suggestion
  bool _stable = true;
```